Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoAddEntity.cpp

```diff
@@ -46,21 +47,50 @@ namespace AzToolsFramework
             PrefabDom parentEntityDomAfterAddingEntity;
             m_instanceToTemplateInterface->GenerateDomForEntity(parentEntityDomAfterAddingEntity, parentEntity);
 
-            PrefabDom& focusedTemplateDom =
-                m_prefabSystemComponentInterface->FindTemplateDom(m_templateId);
-            PrefabDomPath entityPathInFocusedTemplate(parentEntityAliasPath.c_str());
-            // This scope is added to limit their usage and ensure DOM is not modified when it is being used.
+            // If the parent is the focused container entity, we need to fetch from root template rather than
+            // retrieving from the focused template. The focused containter entity DOM in focused template does
+            // not contain transform data seen from root, but after-state entity DOM from serialization always
+            // contain the transform data, which may generate unexpected patches to update transform.
+            if (parentEntityId == focusedInstance.GetContainerEntityId())
             {
-                // DOM value pointers can't be relied upon if the original DOM gets modified after pointer creation.
-                const PrefabDomValue* parentEntityDomInFocusedTemplate = entityPathInFocusedTemplate.Get(focusedTemplateDom);
-                AZ_Assert(parentEntityDomInFocusedTemplate,
-                    "Could not load parent entity's DOM from the focused template's DOM. "
-                    "Focused template id: '%llu'.", static_cast<AZ::u64>(m_templateId));
+                PrefabDom parentContainerEntityDom;
+                m_instanceDomGeneratorInterface->GenerateEntityDom(parentContainerEntityDom, parentEntity);
```

Comment:
yeah...the names are definitely confusing. It took me a while to figure out that GenerateDomForEntity and GenerateEntityDom are actually 2 different functions doing 2 different things. Please add a task to rename these

Reply:
Do you think we still need InstanceDomGenerator? I think these two functions fit better in InstanceToTemplatePropagator. And keeping them near GenerateDomForEntity and GenerateDomForInstance would be good for us to maintain and to be clear about their functionalities imo, or vice versa moving the other two to InstanceDomGenerator.